### DISCUSSION TITLE
RandomCrop in the train pipeline

### DISCUSSION THREAD

**Discussion by junghyun-avikus** [fcakyon/small-object-detection-benchmark#23] [Q&A] [answered]
Hi,

I was reading training config files for several models, and then I saw the following lines.
https://github.com/fcakyon/small-object-detection-benchmark/blob/d345fc03f323778ea3be44b29945c21bde54bcbe/mmdet_configs/visdrone_fcos/fcos_crop_480_960_cls_60.py#L21-L45
It seems like you guys cropped an image of 480x960 then resized it to 1333x800 with the keep_ratio argument set to True.
Then, doesn't it shrink the image because image width(=960) is smaller than target width(=800)?
If my understanding is correct, are there specific reasons for padding so many pixels on height?

My current understanding came from the following RandomCrop implementation by mmdet.
https://github.com/open-mmlab/mmdetection/blob/94afcdc6aa1e2c849c9d8334ec7730bb52b17922/mmdet/datasets/pipelines/transforms.py#L720-L754

Or am I missing something?

**Comment by fcakyon** [comment, marked as answer]
@junghyun-avikus you misunderstood the `crop_size` parameter. It does not crop 480x960 images. I copy this line from the link you provided: `"absolute_range" uniformly samples
            crop_h in range [crop_size[0], min(h, crop_size[1])] and crop_w
            in range [crop_size[0], min(w, crop_size[1])].`

**Reply by junghyun-avikus** [reply]
Oh I missed the crop_type. Sorry for bothering, and thanks for your clarification.

**Reply by fcakyon** [reply]
No problem at all! Have a great day.
